src/hooks/useFetch/useFetch.js

```diff
@@ -0,0 +1,33 @@
+import { useState, useEffect } from 'react';
+
+const useFetch = (url) => {
+  const [data, setData] = useState(null);
+  const [isPending, setIsPending] = useState(true);
+  const [error, setError] = useState(null);
+
+  useEffect(() => {
+    setTimeout(() => {
+      fetch(url)
+      .then(res => {
+        if (!res.ok) { // error coming back from server
+          throw Error('could not fetch the data for that resource');
+        } 
+        return res.json();
+      })
+      .then(data => {
+        setIsPending(false);
+        setData(data);
+        setError(null);
+      })
+      .catch(err => {
+        // auto catches network / connection error
+        setIsPending(false);
+        setError(err.message);
+      })
+    }, 1000);
+  }, [url])
+
+  return { data, isPending, error };
+}
+ 
+export default useFetch;
```

Comment:
Please include a working CodeSandbox.io demo of the same. Refer to existing CodeSandboxes to see how code is structured in there.